Code/Framework/AzCore/AzCore/Serialization/Json/JsonImporter.cpp

```diff
@@ -159,6 +160,311 @@ namespace AZ
+                    }
+                }
+            }
+            else
+            {
+                // Check if the target data already contains a field with the same name
+                auto fieldNameStringRef = rapidjson::Value::StringRefType(fieldName.data(),
+                    static_cast<rapidjson::SizeType>(fieldName.size()));
+                auto targetField = targetDataAtCurrentDepth.FindMember(fieldNameStringRef);
+                // Initialize the Merge Patch result to success
+                // Any recursive calls to ResolveImportsInOrder would be responsible
+                // for converting the result code to failure
+                ResultCode mergePatchResult{ Tasks::Import, Outcomes::Success };
+                if (fieldValue->IsObject() && targetField != targetDataAtCurrentDepth.MemberEnd())
+                {
+                    mergePatchResult.Combine(ResolveImportsInOrder(targetField->value, allocator, *fieldValue,
+                        importPathStack, settings, element));
+                }
+                else if (fieldValue->IsObject() || fieldValue->IsArray())
+                {
+                    // And Object or Array is checked for children $import directives
+                    rapidjson::Value name;
+                    name.CopyFrom(rapidjson::Value(fieldNameStringRef), allocator, true);
+                    rapidjson::Value value;
+                    // Add an empty object field and then use ResolveImportsInOrder to merge to it.
+                    targetDataAtCurrentDepth.AddMember(AZStd::move(name),
+                        rapidjson::Value{ rapidjson::kObjectType }, allocator);
+                    auto newFieldMemberIter = targetDataAtCurrentDepth.FindMember(fieldNameStringRef);
+                    if (newFieldMemberIter == targetDataAtCurrentDepth.MemberEnd())
+                    {
+                        auto reportMessage = ReporterString::format("Failed to Add Member %.*s to JSON while importing.",
+                            AZ_STRING_ARG(fieldName));
+                        if (!importPathStack.empty())
+                        {
+                            reportMessage += ReporterString::format(R"( This occured while importing file "%s".)",
+                                importPathStack.back().c_str());
+                        }
+                        return settings.m_reporting(reportMessage, ResultCode(Tasks::Import, Outcomes::Catastrophic), element);
+                    }
+
+                    mergePatchResult.Combine(ResolveImportsInOrder(newFieldMemberIter->value, allocator, *fieldValue,
+                        importPathStack, settings, element));
+                }
+                else if (fieldValue->IsNull())
+                {
+                    if (targetField != targetDataAtCurrentDepth.MemberEnd())
+                    {
+                        targetDataAtCurrentDepth.RemoveMember(targetField);
+                    }
```

Comment:
could importing/merging `"field":null` could be used to remove a field?

Reply:
Actually yes.
The algorithm here is actually a modified [JSON Merge Patch](https://datatracker.ietf.org/doc/html/rfc7386#section-1) algorithm, that has a special case were if it finds an `"$import"` field, it instead merges the content of the import file into the parent field of the `$import` key.

Reply:
Well it turns out removing null entries with the ResolveImportsInorder causes a problem when reading fields that are legitimately null. 

For example reading in a JSON patch that is trying to set a field to the value of `null` is legitimate use of regular [JSON Patch](https://datatracker.ietf.org/doc/html/rfc6902) (Note: [JSON Merge Patch](https://datatracker.ietf.org/doc/html/rfc7386#section-1) is a different algorithm from [JSON Patch](https://datatracker.ietf.org/doc/html/rfc6902)).

Due to the currently logic fields with null values are being removed before they can be actually merged.
That is causing UnitTest to fail such as this [test](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/Tests/Settings/SettingsRegistryTests.cpp#L682-L691)